### PR TITLE
Array Rotation in Java using Temp Array (Issue #750)

### LINKED ISSUE #750: Array Rotation in c, c++, java, python
Rotate an array of n elements by m elements using a temporary array.

### PR DESCRIPTION
**Pull request by py93**
For the issue #750

Fixes #750

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

- Added Array Rotation by d elements using temporary array in Java

#### Languages Used:
-  Java

#### Files Added:
- ArrayRotationUsingTempArray.java

### CODE REVIEW

**Comment by marwahmanbir** [comment]
Complete the PR template above.

**Comment by py93** [comment]
@marwahmanbir Updated the PR template

File: Array_Rotation/ArrayRotationUsingTempArray.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    private static void rotate(int[] arr, int n, int d) 
+    {
+        //it brings down d to the minimum number of rotations we need to make in case d > n
+         d = d%n; 
+        
+        //initialize temp array of size d
+        int[] temp = new int[d];
+        
+        //copy the first d elements to temp array
+        for(int i=0;i<d;i++)
```
Comment (marwahmanbir) [inline Array_Rotation/ArrayRotationUsingTempArray.java]: Add space after ``for`` and after every ``;``.

File: Array_Rotation/ArrayRotationUsingTempArray.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import java.util.Arrays;
+ 
+public class ArrayRotationUsingTempArray
+{
+    //function to left rotate the array by d elements 
+    private static void rotate(int[] arr, int n, int d) 
+    {
+        //it brings down d to the minimum number of rotations we need to make in case d > n
+         d = d%n;
```
Comment (marwahmanbir) [inline Array_Rotation/ArrayRotationUsingTempArray.java]: Add space around ``%``.

File: Array_Rotation/ArrayRotationUsingTempArray.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        //it brings down d to the minimum number of rotations we need to make in case d > n
+         d = d%n; 
+        
+        //initialize temp array of size d
+        int[] temp = new int[d];
+        
+        //copy the first d elements to temp array
+        for(int i=0;i<d;i++)
+        {
+            temp[i]=arr[i];
```
Comment (marwahmanbir) [inline Array_Rotation/ArrayRotationUsingTempArray.java]: Always add space around operators (+, -, *, /, <, >, <=, >=, &&, ||, etc) and after `for` & `if`.

File: Array_Rotation/ArrayRotationUsingTempArray.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        }
+        
+        //left moves the rest of the elements in original array by d elements
+        for(int i=0;i<n-d;i++)
```
Comment (marwahmanbir) [inline Array_Rotation/ArrayRotationUsingTempArray.java]: Fix spacing.

File: Array_Rotation/ArrayRotationUsingTempArray.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        {
+            arr[i]=arr[i+d];
```
Comment (marwahmanbir) [inline Array_Rotation/ArrayRotationUsingTempArray.java]: Fix spacing.

File: Array_Rotation/ArrayRotationUsingTempArray.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        }
+        
+        //assigns the remaining elements from temp array to original array
+        for(int i=0;i<d;i++)
+        {
+            arr[i+n-d]=temp[i];
```
Comment (marwahmanbir) [inline Array_Rotation/ArrayRotationUsingTempArray.java]: Fix spacing in line 27 & 29.

File: Array_Rotation/ArrayRotationUsingTempArray.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        }
+        
+    }
+     
+    public static void main(String[] args) 
+    {
+        
+        int[] arr = new int[] {1,2,3,4,5,6,7};
```
Comment (marwahmanbir) [inline Array_Rotation/ArrayRotationUsingTempArray.java]: Fix spacing.

File: Array_Rotation/ArrayRotationUsingTempArray.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        int d = 2;
+        System.out.println("Initial Input Array");
+        System.out.println(Arrays.toString(arr));
+        rotate(arr,7, d);
```
Comment (marwahmanbir) [inline Array_Rotation/ArrayRotationUsingTempArray.java]: Space before ``7``

**Comment by py93** [comment]
@marwahmanbir Fixed the spacing.

File: Array_Rotation/ArrayRotationUsingTempArray.java [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+        {
+            arr[i] = arr[i+d];
+        }
+        
+        //assigns the remaining elements from temp array to original array
+        for (int i = 0; i < d; i++)
+        {
+            arr[i+n-d] = temp[i];
+        }
+
```
Comment (GOVINDDIXIT) [inline Array_Rotation/ArrayRotationUsingTempArray.java]: extra line

File: Array_Rotation/ArrayRotationUsingTempArray.java [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+        for (int i = 0; i < d; i++)
+        {
+            arr[i+n-d] = temp[i];
+        }
+        
+    }
+     
+    public static void main(String[] args) 
+    {
+
```
Comment (GOVINDDIXIT) [inline Array_Rotation/ArrayRotationUsingTempArray.java]: extra line

**Comment by py93** [comment]
@GOVINDDIXIT @marwahmanbir Made the changes and updated the PR

**Pull request merged**